LoopKit/GlucoseKit/CachedGlucoseObject+CoreDataClass.swift

```diff
@@ -134,13 +135,17 @@ extension CachedGlucoseObject {
         self.wasUserEntered = sample.wasUserEntered
         self.device = sample.device
         self.trend = sample.trend
+        // The assumption here is that if this is created from a HKQuantitySample, it is coming out of HealthKit, and
+        // therefore, by definition, eligible.
+        self.healthKitEligibleDate = nil
     }
 }
 
 // MARK: - Watch Synchronization
 
 extension CachedGlucoseObject {
     func update(from sample: StoredGlucoseSample) {
+        self.uuid = sample.uuid
```

Comment:
I'm trying to think why I would have left this out. IIRC, I left this out because the watch does not require the `uuid` (nor the new `healthKitEligibleDate`) and it just bloats the phone -> watch data transmission (which is already bad enough).

I think we need to remove both `uuid` and `healthKitEligibleDate` and add a comment why they are not included in Watch transfer.

Reply:
Actually, it was I who _removed_ it.  See https://github.com/tidepool-org/LoopKit/commit/b246ba70eba8bce2501efa7f724aba544c2a3a58#diff-80fb9a5a036254803c7a034e565efa65133808e28a27b036eda28125740b1d73.  That looks wrong to me (I don't know why it passed unit tests...).

Reply:
Ah, yes, okay. Definitely leave it in. I took a quick look around and I now see where the code ensures that `uuid` and `healthKitEligibleDate` **won't** actually be sent to the watch (an unnecessary and large amount of data transfer).